### ISSUE TITLE
50 shades of red

### ISSUE THREAD

**Issue by leeoniya** [krausest/js-framework-benchmark#35]
all red buckets are not created equal, though most green and yellow are. looking at [1], "select row" 13.38ms is the same color as 73.94ms.

it would be useful to have the color scales be tuned to be statistically meaningful within each test. 4300ms and 500ms are very different levels of unacceptable when the baseline is 207.55ms :)

[1] https://cdn.rawgit.com/krausest/js-framework-benchmark/4e47158a10e52122bf4244724cf99b6de4ef7ac1/webdriver-java/table.html

**Comment by krausest** [comment]
Would you like [that](https://cdn.rawgit.com/krausest/js-framework-benchmark/01d859b202efc5f77a0b999781f44adb3f99cfe8/webdriver-java/table.html) better?
Now factors between 1 and 1.5 are absolutely fine and thus colored green,
anything below factor 2.5 is acceptable and thus yellow,
anything else bad which is red.
Yes and why not some shading for in between ;-)
(BTW all values within 1 frame i.e. below 16 msecs are treated equal since I didn't want to penalise frameworks using RAF for repaint.)

**Comment by krausest** [comment]
Or maybe some continuous coloring (factor 2 ist yellow, factor 4 and above is red) like [here](https://cdn.rawgit.com/krausest/js-framework-benchmark/8c2626bfe58de165daa55d8afc5d3381aa7f61c8/webdriver-java/table.html)?

**Comment by leeoniya** [comment]
last version looks great 👍

**Comment by leeoniya** [comment]
So here's a quickie that can be tweaked to adjust the scale. You can run it in console against the generated html table to recolor it. It uses a straight factor without a 16ms threshold, which helps highlight the really fast libs but maybe exaggerates the importance of large factors in sub-frame timings.

```js
// generated via http://www.strangeplanet.fr/work/gradient-generator/?c=16:F44336:FFEB3B:4CAF50
var scale = {
	"#4CAF50": 1.1,
	"#65B74D": 1.2,
	"#7FC04A": 1.3,
	"#98C847": 1.4,
	"#B2D144": 1.5,
	"#CBD941": 1.6,
	"#E5E23E": 1.7,
	"#FFEB3B": 1.8,
	"#FDD63A": 1.9,
	"#FCC139": 2.0,
	"#FAAC39": 2.2,
	"#F99738": 2.4,
	"#F88237": 2.8,
	"#F66D37": 3.2,
	"#F55836": 3.6,
	"#F44336": 4.0,
}

function findColor(factor) {
	for (var col in scale)
		if (factor < scale[col])
			return col;
	return Object.keys(scale).pop();
}

function findBase(means) {
	var min = 1e6;
	for (var j = 0; j < means.length; j++) {
		var mean = +means[j].textContent;
		if (mean < min)
			min = mean;
	}
	return mean;
}

var trs = document.querySelectorAll("tr");

for (var i = 1; i < trs.length; i++) {
	var means = trs[i].querySelectorAll(".mean");
	var base = findBase(means);
	for (var j = 0; j < means.length; j++) {
		var mean = +means[j].textContent;
		var factor = mean / base;
		means[j].parentNode.style.backgroundColor = findColor(factor);
	}
}
```

**Comment by leeoniya** [comment]
if you ignore vanillajs and compare only fastest means between frameworks...
![recolor-chart2](https://cloud.githubusercontent.com/assets/43234/17203400/ea96b76c-5464-11e6-80da-6af8c7e50964.PNG)

**Comment by krausest** [comment]
I'm somehow reluctant to do that for sub 16 msec actions. Here's a picture from vidom:
<img width="786" alt="bildschirmfoto 2016-07-28 um 20 10 55" src="https://cloud.githubusercontent.com/assets/3374055/17224121/db17be58-54ff-11e6-96c5-ce995a6fb5dd.png">
Vidom is a RAF-based framework and would be measured with 11 msecs in that case, since I'm starting with the event and stopping after the paint event finished. The question is whether that is fair if it actually actively consumes 8.8 msecs and waited for the rest. (To be honest: I had to search for such a large gap, but the principle holds). When I developed the benchmark it was important to me to measure the actual duration for an user and not just some javascript duration. And as for the user anything below 16 msecs is simply fast enough. I don't like to penalize RAF based frameworks because using RAF has its share of advantages. The whole issue is negligible for longer running test cases.
